mat64/triangular.go

```diff
@@ -395,6 +396,76 @@ func (t *TriDense) MulTri(a, b Triangular) {
 	}
 }
 
+// Exp calculates the exponential of the matrix a, e^a, placing the result
+// in the receiver. If a has an odd number of rows, the matrix will be converted
+// to a Dense to perform the calculation.
+func (t *TriDense) Exp(a Triangular) {
+	n, kind := a.Triangle()
+	t.reuseAs(a.Triangle())
+
+	aU, _ := untransposeTri(a)
+	t.reuseAs(n, kind)
+	var restore func()
```

Comment:
Okay, that makes sense, but if I run the tests without isolating the work space I get the following error repeated a number of times:
```
list_test.go:925: Original data slice not modified when a maybeSame: *mat64.TriDense.Exp(*mat64.TriDense), size: struct { ar int; ac int }{ar:1, ac:1}, atrans false
```